File: C++/Step12-Dynamic Programming_1/9461.cpp
//
//  9461.cpp
//  Step12-Dynamic Programming_1
//
//  Created by 김예빈 on 2019. 9. 6..
//  Copyright © 2019년 김예빈. All rights reserved.
//

#include <iostream>
using namespace std;

int main(int argc, const char * argv[]) {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int t, n, i;
    long long triangle[100];
    
    triangle[0] = triangle[1] = triangle[2] = 1;
    triangle[3] = triangle[4] = 2;
    triangle[5] = 3;
    
    cin >> t;
    
    while(t--) {
        cin >> n;
        
        for(i = 5; i < n; i++)
            triangle[i] = triangle[i - 1] + triangle[i - 5];
        
        cout << triangle[n - 1] << "\n";
    }
    
    return 0;
}

File: BaekJoon Online Judge/Swift/Step14-Dynamic Programming_1/9461.swift
//
//  9461.swift
//  Step14-Dynamic Programming_1
//
//  Created by Yebin Kim on 2019/12/09.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

var dp = [1, 1, 1, 2, 2]

for i in 5..<100 {
    dp.append(dp[i - 5] + dp[i - 1])
}

if var tc = Int(readLine()!) {
    while tc > 0 {
        let N = Int(readLine()!)!
        print(dp[N - 1])
        
        tc -= 1
    }
}
